### PR TITLE
[MM-55339] Improve types for Markdown component

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Converts Markdown component to functional along with correctly typing it. This is a type fix type I had to do in my other PR, so I just thought to make it into a separate PR

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55339

#### Screenshots
none

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]

**Review by larkox — APPROVED** [review]
LGTM, just one small thing that I have mixed feelings about, but happy to leave it as is.

File: webapp/channels/src/components/markdown/index.ts [hunk 1]
```diff
@@ -14,24 +16,86 @@ import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
...
+import type EmojiMap from 'utils/emoji_map';
+import type {ChannelNamesMap, MentionKey, TextFormattingOptions} from 'utils/text_formatting';
 import {getSiteURL} from 'utils/url';
 
 import type {GlobalState} from 'types/store';
 
 import Markdown from './markdown';
 
-type Props = {
+export type OwnProps = {
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/markdown/index.ts]: Yes you are correct i moved OwnProps inside of the component file instead of index

**Comment by M-ZubairAhmed** [comment]
> Successfully triggered E2E testing! [GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/386210) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/386210)

Shows no failures

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**